src/proofme-utils.ts

```diff
@@ -500,28 +502,39 @@ export class ProofmeUtils {
         }
         // Loop all requested credentials
         for (const requestedCredential of requestedCredentials.credentials) {
-            let isInsideMinimumRequired = false;
-            if (requestedCredentials.minimumRequired) {
-                isInsideMinimumRequired = !!requestedCredentials.minimumRequired.data.find(x => x === requestedCredential.key);
-            }
-            // Check only required keys
-            if (requestedCredential.required && !isInsideMinimumRequired) {
-                if (!Array.isArray(requestedCredential.provider)) {
-                    requestedCredential.provider = [requestedCredential.provider];
-                }
-                let found = false;
-                for (const provider of requestedCredential.provider) {
-                    if (credentials.credentials[provider] &&
-                        credentials.credentials[provider].credentials &&
-                        credentials.credentials[provider].credentials[requestedCredential.key]
-                    ) {
-                        // All good, found!
-                        found = true;
+            if (requestedCredential.required && requestedCredential.provider === "ADDITIONAL_INFO") {
+                const objectKeys = Object.keys(credentials.credentials[requestedCredential.provider].credentials);
+                for (const key of objectKeys) {
+                    const additionalInfo = credentials.credentials[requestedCredential.provider].credentials[key] as IRequestedCredentialKey;
+                    if (additionalInfo.required && (additionalInfo.key as IAdditionalInfo[]).find(x => !x.answer)) {
+                        checkResult.success = false;
+                        checkResult.missingKeys.push(additionalInfo);
                     }
                 }
-                if (!found) {
-                    checkResult.success = false;
-                    checkResult.missingKeys.push(requestedCredential);
+            } else {
+                let isInsideMinimumRequired = false;
+                if (requestedCredentials.minimumRequired) {
+                    isInsideMinimumRequired = !!requestedCredentials.minimumRequired.data.find(x => x === requestedCredential.key);
+                }
+                // Check only required keys
+                if (requestedCredential.required && !isInsideMinimumRequired) {
+                    if (!Array.isArray(requestedCredential.provider)) {
+                        requestedCredential.provider = [requestedCredential.provider];
+                    }
+                    let found = false;
+                    for (const provider of requestedCredential.provider) {
+                        if (credentials.credentials[provider] &&
+                            credentials.credentials[provider].credentials &&
+                            credentials.credentials[provider].credentials[requestedCredential.key as string]
```

Comment:
credentials.credentials[provider]?.credentials?.[requestedCredential.key as string] ?? :)

Reply:
Ja de key interface was verandert dus moest em casten naar een string... :-P. Ik stel voor dat we dit later reworken